andela/pippin-ah-backend
# #162727550 Handle JWT Authentication

#### What does this PR do?
This PR handles JWT authentication for user login/register and route restriction for non-logged in users.

#### Description of Task to be completed?
- Return JWT to User upon successful registration
- Return JWT to User upon successful login
- Restrict non-logged in users from accessing other routes

#### How should this be manually tested?
- Send a POST request to `/api/v1/users/` with user registration details, or
- A POST request to `/api/v1/users/login` with user login details
- You should get a JSON response that includes token
Also,
- Send a get request to `/api/v1/user/:id` without `accesstoken` header or with wrong `accesstoken` in header
- You should get a `401` authorization error

#### What are the relevant pivotal tracker stories?
[#162727550](https://www.pivotaltracker.com/n/projects/2229853/stories/162727550)
